lib/default.nix

```diff
@@ -393,4 +393,96 @@ in {
+          # Also build and cache any tools in the `devShell`
+          devShell = project.shell;
+          # Run HPC on the tests
+          coverage = builtins.listToAttrs (lib.concatMap (packageName: [{
+            name = packageName;
+            value = coverageProject.hsPkgs.${packageName}.coverageReport;
+          }]) (packageNames coverageProject));
+        }
+        # Build the plan-nix and check it if materialized
+        // lib.optionalAttrs (checkedProject ? plan-nix) {
+          plan-nix = checkedProject.plan-nix;
+        }
+        # Build the stack-nix and check it if materialized
+        // lib.optionalAttrs (checkedProject ? stack-nix) {
+          stack-nix = checkedProject.stack-nix;
+        };
+    in {
+      # Used by:
+      #   `nix build .#pkg-name:lib:pkg-name`
+      #   `nix build .#pkg-name:lib:sublib-name`
+      #   `nix build .#pkg-name:exe:exe-name`
+      #   `nix build .#pkg-name:test:test-name`
+      packages = builtins.listToAttrs (
+        lib.concatMap (packageName:
+          let package = project.hsPkgs.${packageName};
+          in lib.optional (package.components ? library)
+                { name = "${packageName}:lib:${packageName}"; value = package.components.library; }
+            ++ lib.mapAttrsToList (n: v:
+                { name = "${packageName}:lib:${n}"; value = v; })
+              (package.components.sublibs)
+            ++ lib.mapAttrsToList (n: v:
+                { name = "${packageName}:exe:${n}"; value = v; })
+              (package.components.exes)
+            ++ lib.mapAttrsToList (n: v:
+                { name = "${packageName}:test:${n}"; value = v; })
+              (package.components.tests)
+            ++ lib.mapAttrsToList (n: v:
+                { name = "${packageName}:bench:${n}"; value = v; })
+              (package.components.benchmarks)
+        ) (packageNames project));
+      # Used by:
+      #   `nix flake check`
+      inherit checks;
+      #   `nix run .#pkg-name:exe:exe-name`
+      #   `nix run .#pkg-name:test:test-name`
+      apps = builtins.listToAttrs (
+        lib.concatMap (packageName:
+          let package = project.hsPkgs.${packageName};
+          in lib.mapAttrsToList (n: v:
+                { name = "${packageName}:exe:${n}"; value = { type = "app"; program = v.exePath; }; })
+              (package.components.exes)
+            ++ lib.mapAttrsToList (n: v:
+                { name = "${packageName}:test:${n}"; value = { type = "app"; program = v.exePath; }; })
+              (package.components.tests)
+            ++ lib.mapAttrsToList (n: v:
+                { name = "${packageName}:benchmark:${n}"; value = { type = "app"; program = v.exePath; }; })
+              (package.components.benchmarks)
+        ) (packageNames project));
+      # Used by hydra.
+      hydraJobs = ciJobs;
+      # Like `hydraJobs` but with system first so that it the IFDs will not have
+      # to form systems we are not testing.
```

Comment:
This comment doesn't make much sense since they're literally the same?

Reply:
Difference is how they are handled by `flake-utils` [here](https://github.com/numtide/flake-utils/blob/04c1b180862888302ddfb2e3ad9eaa63afc60cf8/default.nix#L131-L134).  I'll add a comment.